src/models/data_export.rs

```diff
@@ -0,0 +1,115 @@
+use futures::{future, FutureExt, TryFutureExt};
+use percent_encoding::utf8_percent_encode;
+use serde::Serialize;
+use url::Url;
+
+use crate::{
+    constants::URI_COMPONENT_ENCODE_SET,
+    runtime::{
+        msg::{Action, ActionLoad, Internal, Msg},
+        Effect, EffectFuture, Effects, Env, EnvFutureExt, UpdateWithCtx,
+    },
+    types::{
+        api::{fetch_api, APIRequest, APIResult, DataExportResponse},
+        profile::AuthKey,
+    },
+};
+
+use super::{
+    common::{eq_update, Loadable},
+    ctx::{Ctx, CtxError},
+};
+
+#[derive(Serialize, Default, Clone)]
+#[cfg_attr(debug_assertions, derive(Debug))]
+#[serde(rename_all = "camelCase")]
+pub struct DataExport {
+    /// This is the Loading result of the User data export request.
+    pub export_url: Option<(AuthKey, Loadable<Url, CtxError>)>,
+}
+
+impl DataExport {
+    pub fn new() -> Self {
```

Comment:
use DataExport::default()
we use ::new() only if it returns some effects

Reply:
:+1: fixed.